File: PHP/IntegerListFromUsp/src/Exercicio25.php
<?php

declare(strict_types=1);

namespace Jean\IntegerListFromUsp;

class Exercicio25
{
    public function main(array $params): void
    {
        $ai = 0;
        $bi = 0;
        $a = $params['a'][$ai];
        $b = $params['b'][$bi];
        $this->printf($a, $b);
        $total = 0;
        $soma = 0;
        while ($a != 0) {
            $total += 1;
            $termo = 1;
            for ($i = 1; $i <= $b; $i++) {
                $termo *= $a;
            }
            echo "Resp = $termo\n";
            $soma += $termo;
            echo "Soma = $soma\n";
            $a = $params['a'][++$ai] ?? 0;
            $b = $params['b'][++$bi] ?? 0;
            $this->printf($a, $b);
        }
        echo "Total de pares = $total\n";
    }

    public function printf(int $a, int $b): void
    {
        echo "($a, $b)\n";
    }
}

File: CPP/IntegerListFromUsp/src/Exercicio25.cpp
#include <iostream>
#include <cstdio>

using namespace std;

int main() {
    int pares[][2] = {{2, 3}, {5, 2}, {7, 1}, {0, 5}, {3, 2}};
    int idx = 0;

    int a = pares[idx][0];
    int b = pares[idx][1];
    printf("(%d, %d)\n", a, b);

    int total = 0;
    int soma = 0;
    while (a != 0) {
        total = total + 1;
        int termo = 1;
        for (int i = 1; i <= b; i++) {
            termo = termo * a;
        }
        printf("Resp = %d\n", termo);
        soma = soma + termo;
        printf("Soma = %d\n", soma);
        idx++;
        a = pares[idx][0];
        b = pares[idx][1];
        printf("(%d, %d)\n", a, b);
    }
    printf("Total de pares: %d\n", total);
    return 0;
}
